src/connector.js

```diff
@@ -758,27 +828,57 @@ function updateSearchBoxState( newState ) {
 	}
 
 	if ( lastCharKeyUp === 13 ) {
-		suggestionsElement.hidden = true;
+		closeSuggestionsBox();
 		return;
 	}
 
+	activeSuggestion = 0;
 	if ( !searchBoxState.isLoadingSuggestions && previousState?.isLoadingSuggestions ) {
-		suggestionsElement.textContent = '';
-		searchBoxState.suggestions.forEach( ( suggestion ) => {
-			const node = document.createElement( "li" );
-			node.setAttribute( "class", "suggestion-item" );
-			node.onclick = ( e ) => { 
-				searchBoxController.selectSuggestion(e.currentTarget.innerText);
-				searchBoxElement.value = DOMPurify.sanitize( e.currentTarget.innerText );
-			};
-			node.innerHTML = suggestion.highlightedValue;
-			suggestionsElement.appendChild( node );
-		});
+        suggestionsElement.textContent = '';
+        activeSuggestionWaitMouseMove = true;
+        searchBoxState.suggestions.forEach( ( suggestion, index ) => {
+            const suggestionId = "suggestion-" + ( index + 1 );
+            const node = document.createElement( "li" );
+            node.setAttribute( "class", "suggestion-item" );
+            node.setAttribute( "role", "option" );
+            node.id = suggestionId;
+            node.onmouseenter = ( e ) => {
+                if ( !activeSuggestionWaitMouseMove ) {
+                    activeSuggestion = index + 1;
+                    updateSuggestionSelection();
+                }
+            }
```

Comment:
Add semicolon for readability purposes (as flagged by our build script):
```suggestion
            };
```

Reply:
Done in latest commit